xsd: added XSD package

dorier opened the pull request:
This PR adds a package for XSD CodeSynthesis (https://www.codesynthesis.com/products/xsd/).

dorier commented:
I'm closing this PR, it has an issue. I'll reopen one shortly.

adamjstewart commented on var/spack/repos/builtin/packages/xsd/package.py:
```diff
@@ -0,0 +1,48 @@
+    depends_on('libtool', type='build')
+
+    def install(self, spec, prefix):
+        make()
```
Replace `Package` with `MakefilePackage` in the base class above and remove this line.

dorier replied:
Was I supposed to remove the `make()` line? (I removed the `depends_on('libtool', type='build')`, maybe I shouldn't have?)

adamjstewart commented on var/spack/repos/builtin/packages/xsd/package.py:
```diff
@@ -0,0 +1,48 @@
+        make('install', 'install_prefix='+prefix)
+
+    def setup_environment(self, spack_env, run_env):
+        xercesc_lib_path = "-L"+self.spec['xerces-c'].prefix+"/lib"
```
You should use `self.spec['xerces-c'].libs.search_flags`. You may have to modify the `xerces-c` package to add a `libs` property if the library isn't named `libxerces-c.so`. The idea is that packages should tell other packages where their libraries live instead of making multiple other packages guess. Libraries aren't always installed in `prefix.lib`, as seen by #8797.

adamjstewart commented on var/spack/repos/builtin/packages/xsd/package.py:
```diff
@@ -0,0 +1,48 @@
+
+    depends_on('xerces-c')
+    depends_on('libtool', type='build')
+
```
You'll want to add a `url_for_version` like:
```python
def url_for_version(self, version):
    url = "https://www.codesynthesis.com/download/xsd/{0}/xsd-{1}+dep.tar.bz2"
    return url.format(version.up_to(2), version)
```
This way, Spack will be able to download versions other than 4.0.X if they are ever added.

adamjstewart commented:
```
var/spack/repos/builtin/packages/xsd/package.py:28: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/xsd/package.py:29: [W291] trailing whitespace
var/spack/repos/builtin/packages/xsd/package.py:44: [E226] missing whitespace around arithmetic operator
var/spack/repos/builtin/packages/xsd/package.py:47: [E226] missing whitespace around arithmetic operator
var/spack/repos/builtin/packages/xsd/package.py:47: [E226] missing whitespace around arithmetic operator
```
See http://spack.readthedocs.io/en/latest/contribution_guide.html#flake8-tests

dorier commented:
I've addressed your comments but the CI fails with 2 versions of Python. Each time the tests time out, so I'm under the impression that it's not due to my package.

Is there any way to re-try the failed tests?

adamjstewart commented:
I restarted the failing tests.
